src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1408,6 +1408,67 @@ public void Search(string searchText)
             });
         }
 
+        public void ShowUpdatePackages(string[] updatePackages)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            if (updatePackages == null)
+            {
+                return;
+            }
+
+            if (_topPanel.Filter == ItemFilter.UpdatesAvailable && _updatesTabDataIsLoaded)
+            {
+                SelectMatchingUpdatePackages(updatePackages);
+            }
+            else if (_topPanel.Filter == ItemFilter.Installed && _updatesTabDataIsLoaded)
+            {
+                _topPanel.SelectFilter(ItemFilter.UpdatesAvailable);
+                SelectMatchingUpdatePackages(updatePackages);
+            }
+            else
+            {
+                _topPanel.SelectFilter(ItemFilter.UpdatesAvailable);
+
+                // Hook up an event handler to delay selecting packages until they've all loaded, then unhook
+                // the event handler so it doesn't keep happening on every refresh, only the initial load.
+                EventHandler handler = null;
+                handler = (s, e) =>
+                {
+                    _packageList.LoadItemsCompleted -= handler;
+                    SelectMatchingUpdatePackages(updatePackages);
+                };
+                _packageList.LoadItemsCompleted += handler;
+            }
+        }
+
+        private void SelectMatchingUpdatePackages(string[] updatePackages)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            if (updatePackages == null)
+            {
+                return;
+            }
+
+            // Select all packages if the sentinal * is the first item of the update packages list
+            if ("*".Equals(updatePackages[0], StringComparison.Ordinal))
```

Comment:
I'm not a fan of sentinel values on public APIs. The caller has to know what they are. The fact that you had to document this one line shows it's less clear than the rest of the code here.

In this case it's also possible to break the logic, for example with `["MyPackage", "*"]`.

Could you create a separate public method for updating all packages, or replace `string[]` with a richer type (i.e. a discriminated union).